src/Core/Domain/Carrier/CommandHandler/EditCarrierHandlerInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Carrier\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Command\EditCarrierCommand;
+
+/**
+ * Defines contract for EditCarrierHandler
+ */
+interface EditCarrierHandlerInterface
+{
+    /**
+     * @param EditCarrierCommand $command
+     */
+    public function handle(EditCarrierCommand $command);
```

Comment:
```suggestion
    public function handle(EditCarrierCommand $command): CarrierId;
```
I'm not sure but I think it should be useful to return the carrier ID here, when a new carrier version is created we'll need to get the new ID and redirect to the new page The same thing may also be needed for other commands that modify the carrier like the logo update command, which makes me thing that if we split the carrier edition in too many commands then each command has the opportunity to change (or not) the carrier ID, this could make things complicated when we handled the form update as we'd need extra check to see when/if the carrier has been duplicated

Woudn't it make things simpler if we have a single command to update the carrier? And it can also handle the logo update, and later the range updates as well.